charts/ExploreView.tsx

```diff
@@ -77,34 +72,66 @@ export class ExploreView extends React.Component<ExploreProps> {
     // This is different from the chart's concept of chart type because it includes "WorldMap" as
     // an option, and doesn't include certain chart types we don't support right now, such as
     // scatter plots
-    @observable chartType: ExplorerChartType = ChartType.LineChart
+    @observable.ref chartType: ExplorerChartType = ChartType.LineChart
+
+    // TODO: Inidialize indicatorId from URL query paramter
+    @observable.ref indicatorId?: number = 677
 
     chart: ChartConfig
 
-    dispose!: IReactionDisposer
+    disposers!: IReactionDisposer[]
 
     constructor(props: ExploreProps) {
         super(props)
 
         const chartProps = new ChartConfigProps()
-        extend(chartProps, DUMMY_JSON_CONFIG)
         this.chart = new ChartConfig(chartProps)
+    }
 
-        // We need these updates in an autorun because the chart config objects aren't really meant
-        // to be recreated all the time. They aren't pure value objects and have behaviors on
-        // instantiation that include fetching data over the network. Instead, we rely on their
-        // observable properties, and on this autorun block to connect them to the Explore controls.
-        // -@jasoncrawford 2019-12-04
-        this.dispose = autorun(() => {
-            this.chart.props.type = this.configChartType
-            this.chart.props.hasMapTab = this.isMap
-            this.chart.props.hasChartTab = !this.isMap
-            this.chart.tab = this.tab
-        })
+    componentDidMount() {
+        this.disposers = [
+            // We need these updates in an autorun because the chart config objects aren't really meant
+            // to be recreated all the time. They aren't pure value objects and have behaviors on
+            // instantiation that include fetching data over the network. Instead, we rely on their
+            // observable properties, and on this autorun block to connect them to the Explore controls.
+            // -@jasoncrawford 2019-12-04
+            // Turned the autorun into a reaction in order to make the dependencies (type & isMap) more
+            // explicit -@danielgavrilov 2019-12-09
```

Comment:
Yeah sorry for doing this in this PR, but I was looking at it, and it's really only about excluding `ExplorerView.tab` from the dependent properties... 

`ExploreView.tab` is a scary prop for me because its meaning is "what the tab property of chart config should be set to when the chart type is switched", it's not "the current tab" which I expect. 

I pushed a change to remove the tab property which I think is the better approach here possibly. If you disagree I am happy to return it as it was since it's out of scope for this PR (sorry for starting this whole thing)

Reply:
Not sure why `tab` is scary? If it's just confusing, we could rename it to something more clear?

Not sure what distinction you're making between "what the tab property of chart config should be set to when the chart type is switched" and "the current tab"?

Anyway `tab` is completely a function of `isMap`, which is itself a function of `chartType`, so … I still don't understand the goal of this?

I'll review your new code though.